### ISSUE TITLE
Getting NaN loss

### ISSUE THREAD

**Issue by duncster94** [danielegrattarola/keras-gat#23]
Hello,

After a sufficient number of iterations, my model loss unexpectedly goes to `nan`. I suspect this is to do with the following lines:
https://github.com/danielegrattarola/keras-gat/blob/96916cf4d8dfaa05412e10b11bef9ee4e265d817/keras_gat/graph_attention_layer.py#L127-L128

In my tests so far, when I replace these lines with `dense *= A` I don't get any `nan` loss issues. I think the above masking is resulting in a `-inf` value somewhere that gets turned into a `nan` loss after several downstream computations (though I'm not sure where that is happening).

Is there any reason `dense *= A` won't work instead?

**Comment by duncster94** [comment]
Nevermind, I see. The `-10e9` is used to push the softmax for that element close to `0`. May I suggest replacing line 127 with something like `K.maximum(-10e6 * (1.0 - A), -10e9)` to ensure no `-inf` values are encountered?

**Comment by danielegrattarola** [comment]
Hi, 
thanks for pointing this out. I never encountered the problem myself, but it's probably due to what you suggest. 

If you have tested this out, and seen that it solves the problem, can you open a pull request with the edit?

Thanks, 
Daniele

**Comment by duncster94** [comment]
@danielegrattarola Hey thanks for the quick reply. For a bit more context I'm using these GAT layers to do an unsupervised embedding using an adversarial loss. When I use MSE as my loss I don't have any stability issues. My suggestion above doesn't seem to have fixed the problem (though when I set the mask to something like `-10e4` I can train significantly longer without encountering a `nan` loss).

I'm going to continue searching for a solution and when I find one I'll open a pull request!

**Comment by duncster94** [comment]
Right, so this ended up being due to my training procedure after all, not the GAT. However, trying to use this GAT implementation in the critic of a WGAN-GP isn't possible because the above masking step is causing massive gradients. I'll close this issue and possibly open a new one later. Thanks!

**Comment by danielegrattarola** [comment]
I was looking at the [reference implementation](https://github.com/PetarV-/GAT/blob/master/utils/layers.py) and they don't seem to be doing the masking at all, although it was mentioned in the paper. 
You could try removing the masking on L127-128, but it could impact results. 

Let me know if you find a better solution. 
Cheers, 
Daniele
